src/Model/CompositeTypeGo.cs

```diff
@@ -102,16 +126,88 @@ public CompositeTypeGo(IModelType wrappedType)
         }
 
         /// <summary>
+        /// If PolymorphicDiscriminator is set, makes sure we have a PolymorphicDiscriminator property.
+        /// </summary>
+        private void AddPolymorphicPropertyIfNecessary()
+        {
+            if (!string.IsNullOrEmpty(PolymorphicDiscriminator) && Properties.All(p => p.SerializedName != PolymorphicDiscriminator))
+            {
+                var newProp = base.Add(New<Property>(new

```

Comment:
Why are we saving the result of `base.Add`? It doesn't look like it gets used.

Reply:
Fixed